os_migrate/plugins/module_utils/image.py

```diff
@@ -89,6 +89,14 @@ class Image(resource.Resource):
         'ramdisk_id',
     ]
 
+    # Some parameters are allowed when reading an image resource but not when
+    # creating or updating it.  This list of parameter names is purged from the parameter
+    # list before calling create or update and removed from object properties when reading from sdk
+    # Attribute 'stores' is read-only now. Due to bug https://bugs.launchpad.net/glance/+bug/1889676 glance now
+    # sets stores as a read only property. As such we should remove it from any create or update calls.
+    # An existing pattern is to use the readonly_sdk_params list to do this.
+    readonly_sdk_params = ["store", "stores"]
+
     @classmethod
     def from_sdk(cls, conn, sdk_resource):
         obj = super().from_sdk(conn, sdk_resource)
```

Comment:
We do filtering out of ['properties']['self'] in this function already. This means it never gets serialized out into 'params' in the first place. This should work better than filtering on import, because whatever is in 'params' plays a role in idempotency checking. So if we're not using 'self' or 'store' when creating the resource, it should not be in 'params' at all, as it could be causing "false alarms" when checking whether a resource needs updating in destination cloud or not.

So maybe we can put the variable here and filter out 'self', 'store', 'stores'? (I'm not sure about the 'store' vs. 'stores' difference and if we need both?)